astroplan/core.py

```diff
@@ -649,10 +734,20 @@ def calc_rise(self, time, target, which='nearest', horizon=0*u.degree):
                 return previous_rise
 
         if which == 'nearest':
-            if abs(time - previous_rise) < abs(time - next_rise):
-                return previous_rise
+            if _target_is_vector(target):
```

Comment:
I'm really against sometimes returning a `Time` and sometimes returning a list.  Users will *never* write code that supports that well.  And I think it's generally much better to return a single `Time` object no matter what for broader consistency.  

So as an alternative to `NaN`, What about masked arrays?  Those seem to work with `Time` objects:
```python
>>> np.ma.masked_where([False,True,False], [1,2,3])
masked_array(data = [1 -- 3],
             mask = [False  True False],
       fill_value = 999999)
>>> mjds=np.ma.masked_where([False,True,False], [1,2,3])
>>> t
<Time object: scale='utc' format='mjd' value=[ 1.  2.  3.]>
>>> t=Time(mjds,format='mjd')
>>> t
<Time object: scale='utc' format='mjd' value=[1.0 -- 3.0]>
>>> t.jd
masked_array(data = [2400001.5 -- 2400003.5],
             mask = [False  True False],
       fill_value = 1e+20)
```